src/jit/_typeinfo.h

```diff
@@ -300,8 +304,10 @@ class typeInfo
 
     union {
         CORINFO_CLASS_HANDLE m_cls;
-        // Valid only for type TI_METHOD
+        // Valid only for type TI_METHOD without IsToken
         CORINFO_METHOD_HANDLE m_method;
+        // Valid only for TI_TOKEN with IsToken
+        CORINFO_RESOLVED_TOKEN* m_token;
```

Comment:
A `CORINFO_RESOLVED_TOKEN` is fairly large. Seems like you only ever look at the `hMethod` field. So why not just reuse the `m_method` field for the method handle and use the new token bit to safeguard access?